### DISCUSSION TITLE
Comparing response (status code / body) between payloads

### DISCUSSION THREAD

**Discussion by AlessandroZ** [projectdiscovery/fuzzing-templates#32] [Q&A]
Hello, 

I have some specific needs and I don't know if it could be done using nuclei so maybe you could help me. 

I want to test the reaction of the server using many payloads and doing a diff between answers. 

For example, I have 2 payloads, I test both payloads and then I compare the status code or the number of words of the response body. 

It could be 3 payloads and doing the same comparaison between the first response body and the second one and then depending on the result, I check the difference between the second body and the third one. 

Thanks a lot for your help. 

Have a nice day.

**Comment by AlessandroZ** [comment]
I have found my answer using dsl. Thank you :)

**Comment by AlessandroZ** [comment]
Sorry to bother you again with this question but I didn't find any example to do what I want to do so I'm not sure it"s possible to do it. 

 want to do something similar to this example: 
- https://github.com/projectdiscovery/nuclei-templates/blob/95756197977dec8cbb03f272147d75c5b4f4a05e/http/fuzzing/xff-403-bypass.yaml
```
dsl:
    status_code_1 == 403 && status_code_2 != 403"
```
However using a fuzzing template. 

For example I have my payloads: 
```
   payloads:
      injection:
        - "test1"
        - "test2"
        - "test3"
```
And then I want to compare the difference between these response using something like status_code_1 for test1, etc... 

Right now I think it's not possible to do it but it should be really helpful to detect weird server behavior. 

Thank you for your help.

**Comment by Mzack9999** [comment]
@AlessandroZ I think request history capability still needs to be implemented for fuzzing templates. That's something we are looking forward to adding, along with a comparison with baseline response for differential analysis (the idea would be a standard request to tune the expected answer and then highlight those that are different above a certain threshold from this)

**Comment by AlessandroZ** [comment]
Yes this is exactly what I'm looking for. This feature would be really really helpful for lots of tests. 
I will look forward for this update. 

Thanks a lot.

**Comment by GKNSB** [comment]
Hello, sorry to bring this up after some time, is there any progress on this? Has it been implemented by any chance?
